src/routes/v0/tag/global.rs

```diff
@@ -75,53 +99,77 @@ pub async fn tag_taggers_handler(
     Query(query): Query<TagTaggersQuery>,
 ) -> Result<Json<Vec<String>>> {
     info!(
-        "GET {TAG_TAGGERS_ROUTE} label:{}, reach:{:?}",
-        label, query.reach
+        "GET {TAG_TAGGERS_ROUTE} label:{}, query: {:?}",
+        label, query
     );
 
-    let reach = query.reach;
+    let skip = query.pagination.skip.unwrap_or(0);
+    let limit = query.pagination.limit.unwrap_or(20);
 
-    match TagGlobal::get_tag_taggers(label.clone(), reach).await {
+    match TagGlobal::get_tag_taggers(label.clone(), query.user_id, query.reach, skip, limit).await {
         Ok(Some(post)) => Ok(Json(post)),
         Ok(None) => Err(Error::TagsNotFound { reach: label }),
         Err(source) => Err(Error::InternalServerError { source }),
     }
 }
 
 #[derive(Deserialize)]
-pub struct TagsByReachQuery {
+pub struct TagsByReachPath {
     user_id: String,
     reach: Option<TagStreamReach>,
 }
 
 #[utoipa::path(
     get,
-    path = TAG_REACH_ROUTE,
-    tag = "Global Tags by reach",
+    path = HOT_TAGS_BY_REACH_ROUTE,
+    tag = "Hot Tags by reach",
     params(
         ("user_id" = String, Path, description = "User Pubky ID"),
-        ("reach" = TagStreamReach, Path, description = "Reach type: Follower | Following | Friends")
+        ("reach" = TagStreamReach, Path, description = "Reach type: Follower | Following | Friends"),
+        ("taggers_limit" = Option<usize>, Query, description = "Retrieve N user_id for each tag"),
+        ("skip" = Option<usize>, Query, description = "Skip N tags"),
+        ("limit" = Option<usize>, Query, description = "Retrieve N tag"),
+        ("from" = Option<i64>, Query, description = "Retrieve hot tags from this timestamp"),
+        ("to" = Option<i64>, Query, description = "Retrieve hot tags up to this timestamp"),
+        ("tagged_type" = Option<TaggedType>, Query, description = "Retrieve hot tags by the type of entities tagged with it"),
     ),
     responses(
         (status = 200, description = "Retrieve tags by reach cluster", body = Vec<HotTag>),
         (status = 404, description = "Hot tags not found"),
         (status = 500, description = "Internal server error")
     )
 )]
-pub async fn tags_by_reach_handler(Path(path): Path<TagsByReachQuery>) -> Result<Json<HotTags>> {
+pub async fn tags_by_reach_handler(
```

Comment:
Same as the frontend requested from `/stream` endpoints, we need to merge all of our `hot tags` handlers into one that uses graph or indexes as available for each combination and supports `all|followers|following|friends` and `all|24h|last_month` joint filters.